### ISSUE TITLE
Initial installation on Windows does not include SBOL test suite using pip

### ISSUE THREAD

**Issue by MadraghRua** [SynBioDex/pySBOL2#193]
Hi Folks
I recently updated my pySBOL2 after Jake let the Steering committee know you were ready for release. I followed the pip install instructions and found that although the core pySBOL2 code updated the test cases were not brought through.

In the end I had to do a command line install from a local repo that I had pulled the test suite into using the earlier instructions I got from you. This did work and I was able to start running test cases.

**Comment by tcmitchell** [comment]
Hi @MadraghRua, thanks for the feedback. This is a known issue, see #154. I've got it working locally and it will be part of [1.0b3](https://github.com/SynBioDex/pySBOL2/milestone/6).

Note that the unit tests will only be part of a source install, they won't be part of a normal binary install.

**Comment by MadraghRua** [comment]
OK, then this needs to be spelled out in the docs I think. 
When might 1.0b3 be coming out?
Does this also mean that the tutorial materials will not be included or that you will have some other mechanisms for pulling them in?

**Comment by tcmitchell** [comment]
> OK, then this needs to be spelled out in the docs I think.

That's a good suggestion, thanks. Since it seems to be the norm I hadn't thought of adding a note to the README. We'll include this when the change to include unit tests gets merged.

> When might 1.0b3 be coming out?

Please see the 1.0b3 milestone, linked above, for the current status and expected date of release.

> Does this also mean that the tutorial materials will not be included or that you will have some other mechanisms for pulling them in?

Including the python unit tests doesn't have any bearing on tutorial materials. I don't think we have any plans regarding the tutorial materials one way or the other. Once we can support the tutorial in its entirety we can make a decision on if and how to include the tutorial materials. Tutorial support is covered on issue #14

**Comment by MadraghRua** [comment]
Ok, so what is the best way to get started with using pySBOL2? For instnace I tried the following:
>>>import sbol2
>>>sbol2.testSBOL()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\sbol2\__init__.py", line 43, in testSBOL
    import sbol2.test as unit_tests
ModuleNotFoundError: No module named 'sbol2.test'
>>>help(sbol)
... I get the sbol help and can list the different parts of the package contents and functions.
So that looks good.
Then I try to create a new document:
>>>doc = sbol2.document()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'module' object is not callable
 I can appear to create an object like this:
>>>doc = sbol2.document
But when I start to read in an xml file I get messages like this:
>>> doc.read(r'C:\Users\myuser\Documents\GitHub\pySBOL2\sbol2\test\resources\crispr_example.xml')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
I'm using r in the read statement because python 3 expects the type of data to be raw to read the xml file, otherwise you get unicode error messages.

If you or someone else have the time to either put together some command line examples of how I would open a file, read the various objects and then write them elsewhere I could work with that. Or if you have time for a call to go over how best to get started it would be very appreciated.

Thanks!
Kevin

**Comment by tcmitchell** [comment]
I'm sorry for your trouble. Take a quick look at the [code example](https://github.com/SynBioDex/pySBOL2#code-example) in the README. That should help you out with your attempts to create an empty `Document` and then read an SBOL file. You had the `read` correct from what I can see. So something like this:

```python
import sbol2
doc = sbol2.Document()
doc.read(r'C:\Users\myuser\Documents\GitHub\pySBOL2\sbol2\test\resources\crispr_example.xml')
```

The `r` for raw mode (?) is not related to the contents of the XML file itself. It is the backslash `\` characters in the filename. You could alternatively use double backslashes without the `r`:

```python
>>> crispr_file = 'C:\Users\myuser\Documents\GitHub\pySBOL2\sbol2\test\resources\crispr_example.xml'
  File "<stdin>", line 1
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape
>>> crispr_file = r'C:\Users\myuser\Documents\GitHub\pySBOL2\sbol2\test\resources\crispr_example.xml'
>>> crispr_file
'C:\\Users\\myuser\\Documents\\GitHub\\pySBOL2\\sbol2\\test\\resources\\crispr_example.xml'
>>> crispr_file = 'C:\\Users\\myuser\\Documents\\GitHub\\pySBOL2\\sbol2\\test\\resources\\crispr_example.xml'
```

For `testSBOL()`, it looks like you are using that outside your git clone because I see `Python36\lib\site-packages` in the file path:

```python
>>>sbol2.testSBOL()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\sbol2\__init__.py", line 43, in testSBOL
    import sbol2.test as unit_tests
ModuleNotFoundError: No module named 'sbol2.test'
```

That's due to the known issue (#154) that tests are not yet included in the distribution.

**Comment by tcmitchell** [comment]
@MadraghRua this issue is going to get closed when pull request #194 is merged. We still want to hear from you and help you. Your feedback is valuable at this stage in our beta testing. Please continue on here, or open another issue, whichever makes more sense for you.

**Comment by tcmitchell** [comment]
Hi @MadraghRua sorry I didn't notify you sooner. We released [1.0 beta 3](https://github.com/SynBioDex/pySBOL2/milestone/6). Would you be able to try it out and provide feedback about the inclusion of the tests? Both the [README](../blob/master/README.md) and [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md) have info about how to run the tests, depending on how you want to do the installation.

I have reopened this issue so that you can provide feedback. Thanks!

**Comment by tcmitchell** [comment]
Having heard no feedback I'll close this issue out again. Feel free to reopen it if you have any feedback. Thanks!
